src/uu/factor/src/numeric/gcd.rs

```diff
@@ -72,10 +72,14 @@ mod tests {
             gcd(1, a) == 1
         }
 
-        fn divisor(a: u64, b: u64) -> bool {
+        fn zero(a: u64) -> bool {
+            gcd(0, a) == a
+        }
+
+        fn divisor(_a: u64, _b: u64) -> bool {
             // Test that gcd(a, b) divides a and b
             let g = gcd(a, b);
-            a % g == 0 && b % g == 0
+            g == 0 || (a % g == 0 && b % g == 0)
```

Comment:
I understand your concern about adding unnecessary complexity, but this formulation allows falsely passing tests.

For example, `divisor(a, b)`, with `a` or `b` non-zero, will incorrectly "pass" if `gcd()` incorrectly returns a `0` result. So, either the quickcheck input range needs to be restricted or the special case needs to be handled directly. 

From my reading of the quickcheck docs, range restriction will add as much or more complexity to the function.

If you hate the `if ...` formulation (2a50dc42c146cd89c58eaace7750961551b7aee8), an alternative would be...

```suggestion
            (g == 0 && a == 0 && b == 0) || (g != 0 && a % g == 0 && b % g == 0)
```

That should make the test correct for all inputs, and I'm ok with it. But I do think that the `if ...` formulation is easier to understand.